src/measures/fermions/stag.cpp

```diff
@@ -212,24 +213,43 @@ namespace nissa
       generate_fully_undiluted_eo_source(src,noise_type,twall);
     }
     
+	void local_trace(complex* point_result, eo_ptr<color> A, eo_ptr<color> B)
+{
+    //compute results for single points
+    vector_reset(point_result);
+    for(int par=0;par<2;par++)
+        NISSA_PARALLEL_LOOP(ieo,0,locVolh)
+            for(int ic=0;ic<3;ic++)
+                complex_summ_the_conj1_prod(point_result[loclx_of_loceo[par][ieo]],A[par][ieo][ic],B[par][ieo][ic]);
+    NISSA_PARALLEL_LOOP_END;
+    THREAD_BARRIER();
+}
     //take the trace between A^dag and B
     void summ_the_trace(double* out,complex* point_result,eo_ptr<color>  A,eo_ptr<color>  B)
     {
       
       //compute results for single points
-      vector_reset(point_result);
-      for(int par=0;par<2;par++)
-	NISSA_PARALLEL_LOOP(ieo,0,locVolh)
-	  for(int ic=0;ic<3;ic++)
-	    complex_summ_the_conj1_prod(point_result[loclx_of_loceo[par][ieo]],A[par][ieo][ic],B[par][ieo][ic]);
-      NISSA_PARALLEL_LOOP_END;
-      THREAD_BARRIER();
+      local_trace(point_result, A, B);
       
       //final reduction
       complex temp;
       glb_reduce(&temp,point_result,locVol);
       if(IS_MASTER_THREAD) complex_summassign(out,temp);
     }
+	
+	void summ_the_time_trace(double* out,complex* point_result,eo_ptr<color>  A,eo_ptr<color>  B)
+    {
+      
+      //compute results for single points
+      local_trace(point_result, A, B);
+      
+      // reduction over 3-spatial-volume
+      complex temp[glbSize[0]];
+      glb_reduce(temp,point_result,locVol,glbSize[0],locSize[0]);	//(ask sunpho)check if the parameters passed for the reduction over space are correct
```

Comment:
you have to add ,glbCoordOfLoclx[0][0] as a last argument as in https://github.com/sunpho84/nissa/blob/827ca27f9c794b9a5c3eb0eab50aaf2a2c09eca2/src/measures/fermions/mesons.cpp#L116